test/jdk/java/lang/Thread/virtual/stress/GetStackTraceALotWhenPinned.java

```diff
@@ -26,30 +26,44 @@
  * @bug 8322818
  * @summary Stress test Thread.getStackTrace on a virtual thread that is pinned
  * @requires vm.debug != true
- * @run main GetStackTraceALotWhenPinned 25000
+ * @modules java.base/java.lang:+open
+ * @library /test/lib
+ * @run main GetStackTraceALotWhenPinned 500000
  */
 
 /*
  * @test
  * @requires vm.debug == true
- * @run main/timeout=300 GetStackTraceALotWhenPinned 10000
+ * @modules java.base/java.lang:+open
+ * @library /test/lib
+ * @run main/timeout=300 GetStackTraceALotWhenPinned 200000
  */
 
 import java.time.Instant;
 import java.util.concurrent.atomic.AtomicInteger;
 import java.util.concurrent.locks.LockSupport;
+import jdk.test.lib.thread.VThreadRunner;
 
 public class GetStackTraceALotWhenPinned {
 
     public static void main(String[] args) throws Exception {
-        var counter = new AtomicInteger(Integer.parseInt(args[0]));
+        // need at least two carrier threads when main thread is a virtual thread
+        if (Thread.currentThread().isVirtual()) {
+            VThreadRunner.ensureParallelism(2);
```

Comment:
Given that this changes the JVM level scheduler's parallelism, should we now use `/othervm` for these tests to prevent this change interfering with other tests?

Reply:
The ensureParallelism here is to allow the test run on a container configured with a single CPU. I don't think it will impact any other tests that run in the same agent VM, at least I couldn't find any tests where it would create an issue. But you are right that it would be better to restore it or run the test in /othervm mode. The equivalent test in the loom repo does run in /othervm mode as it has to launch with some VM options so we can keep it consistent with that. Thanks.